### PR TITLE
Added Certification Profile link and frontend view

### PR DESCRIPTION
**Pull request by yvesh**
Pull Request for issue #141 

Register + My Profile view:

Frontend View:

Ping @sam-suresh @jeckodevelopment @sanderpotjer @Joomv

P.S.: @sanderpotjer I added the default htaccess.txt back, makes installing easier.

### CODE REVIEW

**Comment by sam-suresh** [comment]
This is great. Thank you.

**Comment by jeckodevelopment** [comment]
@yvesh this is really AWESOME!!!!
even better than my expectation!
Thank you!

File: www/components/com_volunteers/views/volunteer/tmpl/default.php [hunk 1]
```diff
@@ -23,6 +23,14 @@
 <div class="row-fluid profile">
     <div class="span3 volunteer-image">
 		<?php echo VolunteersHelper::image($this->item->image, 'large', false, $this->item->name); ?>
+
+	    <?php if ($this->item->certification): ?>
+		    <div class="volunteer-certificated">
+			    <a href="https://exam.joomla.org/directory/user/<?php echo($this->item->certification) ?>" target="_blank">
+	                <img src="https://exam.joomla.org/images/badge.png" class="img-responsive" />
```
Comment (zero-24) [inline www/components/com_volunteers/views/volunteer/tmpl/default.php:30]: tabs please :P

Comment (zero-24) [reply www/components/com_volunteers/views/volunteer/tmpl/default.php:30]: maybe we should also rename this thing on the server to something more usefull as in theorie there could be more than one certificat

**Comment by zero-24** [comment]
As noted inline this works for one zertifikat but not for more ;)

**Comment by jeckodevelopment** [comment]
@zero-24 we have just one exam at the moment. 
When we'll have more exams and more certificates, we'll look at that :)

File: www/components/com_volunteers/views/volunteer/tmpl/default.php (hunk 1 above)
Comment (wilsonge) [reply www/components/com_volunteers/views/volunteer/tmpl/default.php:30]: +1

**Comment by sam-suresh** [comment]
Even if a user has more than one certification (Joomla!) the profile address will remain same.

**Comment by zero-24** [comment]
> Even if a user has more than one certification (Joomla!) the profile address will remain same.

sure. But that thing is not in question ;) It is about the badge.

And no one is checking that it is my and a valid / passed account. I could add some random text and the badge shows up.

For the other buttons i think this is fine but showing a "you are zertified" should invlove some checking expecial as this is a service hosted by us.

**Comment by sam-suresh** [comment]
No. I got it. It can be misused.

**Comment by zero-24** [comment]
> Possible to add if it check and verify the link is not broken before allow to save?

I can add the link to yves profile in my own.

Also this does not fix the problem of having more than one cert and i have maybe passed the 4.0 one and not the 3.x one.

IMO the exam site needs a real api where you can get the data and check them. And also get a list of certs they have successull passed. On the portal we need a new tab or similiar that lists all certs.

**Comment by sam-suresh** [comment]
One  workaround is to display the user's real name taken from certified directory and display it right under the image. 

Or I can provide a JSON output from exam site which this feature should validate existence of the email address before displaying in this profile. If doesnt exist, it shouldnt display the image.

**Comment by zero-24** [comment]
> Or I can provide a JSON output from exam site which this feature should validate existence of the email address before displaying in this profile. If doesnt exist, it shouldnt display the image.

This is a step in the right direction.

Can you also provide a api for the list of certs a user have?

**Comment by yvesh** [comment]
With everything in the VP, also the other data, there is the possibility of fraud. Not sure how far we should go to protect for that. 

There is a Report as Spammer button and i am sure it's fastly seen when people start faking certificate links.. there is not much gain for them in anyways.

> Can you also provide a api for the list of certs a user have?

@zero-24 Doing a mapping between VP + exams user isn't easy.. How would you query that? You can't rely name, email etc. matches.. the exams site for that would need to store the user id from the VP

**Comment by zero-24** [comment]
> You can't rely name, email etc. matches.. the exams site for that would need to store the user id from the VP

Sounds like a plan 😄 Or do you see a problem here with the ID thing? A other way arrround could be that the exam site generate a random code that you need to enter and with that code the verification is done (that would allow to reuse the api for other things too) For sure that code should be hold private and *not* accessible by anyone expeted user with DB access and the user itself. I'm excluding here explizite the admins and other privileged persons on the portal ;)

**Comment by yvesh** [comment]
> Sounds like a plan :smile: Or do you see a problem here with the ID thing?

Yeah someone needs to manually update all people that made the exam :-) As i said i am not sure it's worth it..

I personally think there isn't a big chance of fraud (that would go unnoticed), you can easily filter all people with exams link and than see if they match and if not ban the spammer. On the long term a good validation system would make sense, but for now it's probably out of scope and not needed. (Just my personal opinion comparing workload vs use).

**Comment by zero-24** [comment]
> Yeah someone needs to manually update all people that made the exam :-) As i said i am not sure it's worth it..

So why do you don't want to go with the `random code` generated by the user on the exam site? (Like all other services like google etc. do that too ;)) This has the advantage that the api can be implemented used also at other places when needed.

**Comment by sanderpotjer** [comment]
I think this is a great first step, so will merge it now. If another type of API will be available in the future for the exam site we can update the integration. 

Thanks @yvesh for the PR!

**Comment by zero-24** [comment]
Looks like I'm the only on who think we should take that exam thing seriously In order to not damage its reputation.

Expecialy as a API was proposed by @sam-suresh already..

**Comment by sam-suresh** [comment]
I would like to see its merged now. Although there is API, it can be difficult to query like outlined by @yvesh. 

The exam site is Open Badges compatible and the badges are hashed uniquely. We can add a Joomla! module to display the badges in the future. I will add this for future improvement.

**Comment by jeckodevelopment** [comment]
Thanks everyone :)

**Pull request merged**